### ISSUE TITLE
Error compiling example from README.md

### ISSUE THREAD

**Issue by akamaus** [Nycto/AStarNim#2]
```
nim   c --run  astar_tst.nim                                                                                                                   
Hint: used config file '/home/dvyal/tmp/nim-0.14.2/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: astar_tst [Processing]
Hint: astar [Processing]
Hint: binaryheap [Processing]
Hint: strutils [Processing]
Hint: parseutils [Processing]
Hint: math [Processing]
Hint: tables [Processing]
Hint: hashes [Processing]
Hint: etcpriv [Processing]
Hint: options [Processing]
Hint: typetraits [Processing]
astar_tst.nim(46, 18) Error: type mismatch: got (Grid) but expected 'Graph = astar'
```

```
% nim -v                                                                                                                                          
Nim Compiler Version 0.14.2 (2016-06-11) [Linux: amd64]
Copyright (c) 2006-2016 by Andreas Rumpf

git hash: abf4e204c2f1a8a6915a8d32c970b5ce64f58e10
active boot switches: -d:release
```

**Comment by Nycto** [comment]
Looks like something changed in the Nim compiler to break this. A rebuild of trunk is now failing, despite passing when originally committed.

**Comment by Nycto** [comment]
I ran a git bisect on Nim. Looks like this commit is the culprit:

https://github.com/nim-lang/Nim/commit/e04f3195407cc99f958dd81e0a9d58fe5414e631

**Comment by akamaus** [comment]
That's weird, apart breaking astar that commit says it fixes the issue 4084, but looks like it isn't fixed :(

**Comment by Nycto** [comment]
I did some fiddling this morning and found that explicitly importing the `hashes` module in the client code resolves this: 5f885c42a6bce5aa9bfdcc5907a6a16f622ce6b6. It's a bit of an arbitrary change, but it seems to solve the problem.

That commit also changes references from `THash` to just `Hash` as it looks like that is a thing now: https://github.com/nim-lang/Nim/commit/6914244f3054a60b7f12622c8d5fbe5e880f7c56
